src/cpu/instruction_set/instruction_set.cpp

```diff
@@ -12,6 +12,7 @@
 
 #include "infoware/cpu.hpp"
 #include "infoware/detail/cpuid.hpp"
+#include <vector>
```

Comment:
split this file into `instruction_set_x86.cpp` with a `#if INFOWARE_X86` at the top and put the new implementation into `instruction_set_darwin_arm.cpp` with `#if __APPLE__ && INFOWARE_ARM`